### PR TITLE
add lineage_qc dockerfile, install nextclade

### PR DESCRIPTION
**Pull request by phoenixAja**
### Summary:
- **What:** add a new dockerfile for nextclade and pangolin, leave the old pangolin image for now until we're ready to switch it over. 
- **Ticket:**  [sc217893](https://app.shortcut.com/genepi/story/217893)
- **Env:** `<rdev link>`

### Checklist:
- [x] I merged latest `<base branch>`
- [x] I manually verified the change
- [x] I added labels to my PR
- [ ] I tested in multiple browsers
- [ ] I added relevant unit tests
- [ ] I have notified others of changes they need to make locally (migrations, jobs, package updates, etc)

### CODE REVIEW

File: src/backend/Dockerfile.lineage_qc
Comment (jgadling) [inline src/backend/Dockerfile.lineage_qc]: Can you use python:3.10-slim instead? I've been meaning to upgrade the other images now that nextstrain is on a newer python, but haven't had a chance. At least if we build this one on the newer version we won't have to worry about fixing it when we upgrade.

File: src/backend/install_nextclade.sh
Comment (jgadling) [inline src/backend/install_nextclade.sh]: Any reason we're using conda to install nextclade instead of using their cli-download option?
https://docs.nextstrain.org/projects/nextclade/en/stable/user/nextclade-cli.html#download-from-command-line

File: .github/workflows/push-tests.yml
Comment (jgadling) [inline .github/workflows/push-tests.yml:222]: Note - this means we'll need to add the genepi-lineage-qc ECR repository to genepi-infra *before* this pr gets merged.

**Review by jgadling — CHANGES_REQUESTED** [review]
Looks good overall - I'd prefer to avoid using conda as much as possible, and I'm flagging this as "don't merge" until we get the new ECR created.

File: src/backend/Dockerfile.lineage_qc
Comment (phoenixAja) [reply src/backend/Dockerfile.lineage_qc]: oh sure!

File: src/backend/install_nextclade.sh
Comment (phoenixAja) [reply src/backend/install_nextclade.sh]: i thought the conda install was cleaner than using curl, plus we already have to install miniconda for pangolin so might as well

File: .github/workflows/push-tests.yml
Comment (phoenixAja) [reply .github/workflows/push-tests.yml:222]: i think we do have that ecr repository created already [here](https://github.com/chanzuckerberg/genepi-infra/blob/9affbe5cd2da4db089a5527f340d1dfb18eca67e/terraform/modules/genepi/main.tf#L32)

**Comment by phoenixAja** [comment]
i disagree on not using conda, to me it's a nice package manager with little overhead, and it seems cleaner to use conda to install rather than using curl

File: src/backend/install_nextclade.sh
Comment (jgadling) [reply src/backend/install_nextclade.sh]: I'd really prefer to avoid using conda as much as possible. If we get into a situation where we just *have* to manage conda environments due to conflicting package requirements, we can deal with it then, but in the meantime the curl installation is pretty clean (it's how we install miniconda already 😢 )

File: src/backend/Dockerfile.lineage_qc
Comment (phoenixAja) [reply src/backend/Dockerfile.lineage_qc]: hmm that seems to break the poetry install..

Comment (phoenixAja) [reply src/backend/Dockerfile.lineage_qc]: we might have to wait on upgrading for now until we have more time to debug if that's ok, maybe it can be a follow up PR

**Comment by phoenixAja** [comment]
@jgadling i updated based on your feedback. I used curl to install nextclade and also updated the base image to python:3.10-slim

**Review by jgadling — APPROVED** [review]

**Pull request merged**
